src/xmipp/libraries/reconstruction_starpu/reconstruct_fourier_codelets.cpp

```diff
@@ -0,0 +1,171 @@
+	padded_image_to_fft.modes[2] = STARPU_SCRATCH; // Raw FFT Scratch Area
+	padded_image_to_fft.modes[3] = STARPU_R; // LoadedImagesBuffer
+	padded_image_to_fft.specific_nodes = 1;
+	padded_image_to_fft.nodes[0] = STARPU_SPECIFIC_NODE_LOCAL;
+	padded_image_to_fft.nodes[1] = STARPU_SPECIFIC_NODE_LOCAL;
+	padded_image_to_fft.nodes[2] = STARPU_SPECIFIC_NODE_LOCAL;
+	padded_image_to_fft.nodes[3] = STARPU_SPECIFIC_NODE_CPU;
+	padded_image_to_fft.name = "codelet_padded_image_to_fft";
+	// Not using any model for this, because CUDA codelet has huge initialization time, but then is very fast. This skews all statistics and decisions are not very meaningful.
+	//padded_image_to_fft.model = &padded_image_to_fft_model;
+	// cl_arg: PaddedImageToFftArgs
+
+	static struct starpu_perfmodel reconstruct_fft_model;
+	reconstruct_fft_model.type = STARPU_REGRESSION_BASED;
+	reconstruct_fft_model.symbol = "reconstruct_fft_model";
+
+	// Reconstruct FFT Codelet
+	reconstruct_fft.where = STARPU_CPU | STARPU_CUDA;
+	// NOTE: From StarPU/examples/cg/cg_kernels.c it seems that STARPU_SPMD applies only to CPU implementations,
+	//       which we rely on. However, the documentation does not say that explicitly, so beware.
+	reconstruct_fft.type = starpu_codelet_type::STARPU_SPMD;
+	reconstruct_fft.cpu_funcs[0] = func_reconstruct_cpu_lookup_interpolation;
+	reconstruct_fft.cpu_funcs[1] = func_reconstruct_cpu_dynamic_interpolation; // (Typically about 2x slower than lookup interpolation)
+	reconstruct_fft.cpu_funcs_name[0] = "func_reconstruct_cpu_lookup_interpolation";
+	reconstruct_fft.cpu_funcs_name[1] = "func_reconstruct_cpu_dynamic_interpolation";
+	reconstruct_fft.cuda_funcs[0] = func_reconstruct_cuda;
+	reconstruct_fft.cuda_flags[0] = STARPU_CUDA_ASYNC;
+	reconstruct_fft.nbuffers = 6;
+	reconstruct_fft.modes[0] = STARPU_R; // FFT Buffer
+	reconstruct_fft.modes[1] = STARPU_R; // Traverse Spaces Buffer
+	reconstruct_fft.modes[2] = STARPU_R; // Blob Table Squared Buffer (only present if fastLateBlobbing is false)
+	reconstruct_fft.modes[3] = STARPU_REDUX; // Result Volume Buffer
+	reconstruct_fft.modes[4] = STARPU_REDUX; // Result Weights Buffer
+	reconstruct_fft.modes[5] = STARPU_R; // LoadedImagesBuffer
+	reconstruct_fft.specific_nodes = 1;
+	reconstruct_fft.nodes[0] = STARPU_SPECIFIC_NODE_LOCAL;
+	reconstruct_fft.nodes[1] = STARPU_SPECIFIC_NODE_LOCAL;
+	reconstruct_fft.nodes[2] = STARPU_SPECIFIC_NODE_LOCAL;
+	reconstruct_fft.nodes[3] = STARPU_SPECIFIC_NODE_LOCAL;
+	reconstruct_fft.nodes[4] = STARPU_SPECIFIC_NODE_LOCAL;
+	reconstruct_fft.nodes[5] = STARPU_SPECIFIC_NODE_CPU;
+	reconstruct_fft.name = "codelet_reconstruct_fft";
+	reconstruct_fft.model = &reconstruct_fft_model;
+	// cl_arg: ReconstructFftArgs
+
+	// Redux volume & weights
+	// Init volume
+	static struct starpu_perfmodel redux_init_volume_model;
+	redux_init_volume_model.type = STARPU_REGRESSION_BASED;
+	redux_init_volume_model.symbol = "redux_init_volume_model";
```

Comment:
Do you mean generally for the `redux_init_volume_model` or for the `"redux_init_volume_model"` string? Because the string is arbitrary and never referred to anywhere else again (StarPU uses it to identify the performance model for saving into files). It is just a coincidence that the variable has the same name as the string.